apps/sms/test/unit/navigation_test.js

```diff
@@ -521,6 +521,66 @@ suite('navigation >', function() {
     });
   });
 
+  suite('ensureCurrentPanel >', function() {
+    setup(function(done) {
+      this.sinon.stub(Navigation, 'slide').returns(Promise.resolve());
+      this.sinon.spy(Navigation, 'toPanel');
+
+      // Set custom hash to have non-initialized current panel
+      window.location.hash = '#x-unknown-panel';
+
+      Navigation.init().catch((e) => {
+        Navigation.toPanel.reset();
+
+        // It's expected error for the unknown panel
+        if (e.message === 'Panel x-unknown-panel is unknown.') {
+          return;
+        }
+
+        return Promise.reject(e);
+      }).then(done, done);
+    });
+
+    test('navigates to default panel when current panel is not set',
+    function(done) {
+      // We don't have any correct current panel, so navigation to default
+      // panel should be triggered.
+      Navigation.ensureCurrentPanel();
+
+      sinon.assert.calledOnce(Navigation.toPanel);
+      sinon.assert.calledWith(Navigation.toPanel, 'panel1');
```

Comment:
instead of looking whether internal methods are called, maybe we can return a promise from ensureCurrentPanel like you suggested, and then check whether the current panel is 'panel1' using the public method isCurrentPanel ?